### PR TITLE
feat: Separate access controls for guests ands services SQSERVICES-1052

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1052" title="SQSERVICES-1052" target="_blank">StorySQSERVICES-1052</a>  [iOS] Separate access controls for guests and services
  </td></table>
  <br />
 

----
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Initially this issue came up by trying to run some Integration Tests on SE using an m1 machine. With the work that Bill started we're able to use the UTIHelper instead of the deprecating methods. That seems to fix the issues on SE while running the tests on an m1 machine. 

In addition to that I added support for accessRoleV2 in Mock Conversation. That's part of the work for separating access controls for guests and services.

--
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by sb88k — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

File: Tests/Source/Conversations/MockTransportSessionConversationsTests.swift [hunk 1]
```diff
@@ -100,39 +100,50 @@ class MockTransportSessionConversationsTests_Swift: MockTransportSessionTests {
     func testThatPushPayloadIsPresentWhenChangingAccessMode() {
         // given
         let newAccessMode = ["invite", "code"]
+        let newAccessRoleV2 = ["team_member", "non_team_member"]
```
Comment (KaterinaWire) [inline Tests/Source/Conversations/MockTransportSessionConversationsTests.swift:103]: Shouldn't be `["team_member", "non_team_member", "guest"]` ?

File: Source/Conversations/MockConversation.swift [hunk 2]
```diff
@@ -51,42 +51,49 @@ extension MockConversation {
...
 
     @objc var changePushPayload: [String: Any]? {
         let accessModeKeyPath = #keyPath(MockConversation.accessMode)
         let accessRoleKeyPath = #keyPath(MockConversation.accessRole)
+        let accessRoleV2KeyPath = #keyPath(MockConversation.accessRoleV2)
 
-        if changedValues()[accessModeKeyPath] != nil || changedValues()[accessRoleKeyPath] != nil {
-            return [ "access_role" : self.accessRole, "access" : self.accessMode ]
+        if changedValues()[accessModeKeyPath] != nil || changedValues()[accessRoleKeyPath] != nil, changedValues()[accessRoleV2KeyPath] != nil {
+            return [ "access_role" : self.accessRole, "access_role_v2": self.accessRoleV2, "access" : self.accessMode ]
```
Comment (KaterinaWire) [inline Source/Conversations/MockConversation.swift:96]: Not sure if it's possible to have in one payload `accessRole` and `accessRoleV2`

Comment (agisilaos) [reply Source/Conversations/MockConversation.swift:96]: For now we do until I finish with the mapping between old and new values.

**Review by KaterinaWire — APPROVED** [review]

**Comment by agisilaos** [comment]
release major

**Pull request merged**